pygpu/gpuarray.pyx

```diff
@@ -955,40 +935,35 @@ cdef class GpuContext:
     """
     def __dealloc__(self):
         if self.ctx != NULL:
-            self.ops.buffer_deinit(self.ctx)
+            gpucontext_deref(self.ctx)
 
     def __reduce__(self):
         raise RuntimeError, "Cannot pickle GpuContext object"
 
-    def __cinit__(self, kind, devno, int flags):
+    def __cinit__(self, bytes kind, devno, int flags):
         cdef int err = GA_NO_ERROR
-        cdef void *ctx
-        self.ops = get_ops(_s(kind))
-        self.ctx = self.ops.buffer_init(devno, flags, &err)
+        cdef gpucontext *ctx
+        self.kind = kind
+        self.ctx = gpucontext_init(<char *>self.kind, devno, flags, &err)
         if (err != GA_NO_ERROR):
             if err == GA_VALUE_ERROR:
                 raise get_exc(err), "No device %d"%(devno,)
             else:
-                raise get_exc(err), self.ops.ctx_error(NULL) + ": " + str(devno)
+                raise get_exc(err), gpucontext_error(NULL, err).decode('utf-8') + ": " + str(devno)
 
     def __enter__(self):
         if cuda_enter == NULL:
             raise RuntimeError("cuda_enter not available")
         if cuda_exit == NULL:
             raise RuntimeError("cuda_exit not available")
-        if self.ops != gpuarray_get_ops("cuda"):
+        if self.kind != b"cuda":
```

Comment:
Why use bytes for the kind? I saw many changes at many places related to that and it seem to make the code more ugly. Can we avoid that?